src/color-scale/color-scale.js

```diff
@@ -17,31 +20,253 @@ const Self = class ColorScale extends ColorElement {
+		};
 	}
 
 	connectedCallback () {
 		super.connectedCallback?.();
+		this._el.swatches.addEventListener("labelchange", this, {capture: true});
 		this._el.swatches.addEventListener("colorchange", this, {capture: true});
+		this._el.swatches.addEventListener("click", this, {capture: true});
+		this._slots.add_button.addEventListener("click", this);
 	}
 
 	disconnectedCallback () {
 		this.#swatches = [];
+		this._el.swatches.removeEventListener("labelchange", this, {capture: true});
 		this._el.swatches.removeEventListener("colorchange", this, {capture: true});
+		this._el.swatches.removeEventListener("click", this, {capture: true});
+		this._slots.add_button.removeEventListener("click", this);
 	}
 
 	handleEvent (event) {
+		let source = event.target;
+
+		if (event.type === "click" && source === this._el.add_button || this._slots.add_button.assignedElements().includes(source)) {
+			this.addColor();
+			return;
+		}
+
+		if (event.type === "colorchange" && source.matches("color-swatch:not(.intermediate, .ignore-updates)")) {
+			this.updateColor(source);
+		}
+		else if (event.type === "labelchange" && source.matches("color-swatch:not(.intermediate, .ignore-updates)")) {
+			this.updateColorName(source);
+		}
+		else if (event.type === "click" && source.closest("button[part=delete-button]")) {
+			this.deleteColor(source.closest("color-swatch"));
+		}
+
 		this.dispatchEvent(new event.constructor(event.type, {...event}));
 	}
 
 	propChangedCallback ({name, prop, detail: change}) {
-		if (name === "computedColors") {
+		if (name === "computedColors" && !this.editable?.name && !this.editable?.color) {
 			// Re-render swatches
+			// Only if nothing is being edited, otherwise the input would be lost
+			// or, e.g., "red" would be converted to "rgb(100%, 0%, 0%)" right after the typing is done
+			this.render();
+		}
+
+		if (name === "editable") {
+			if (this.editable?.add) {
+				if (!this._el.add_button) {
+					this._el.add_button = Object.assign(document.createElement("button"), {part: "add-button", title: "Add color", textContent: "+"});
+					this._slots.add_button.append(this._el.add_button);
+				}
+			}
+			else {
+				this._el.add_button?.remove();
+				this._el.add_button = null;
+			}
+
 			this.render();
 		}
 	}
 
 	#swatches = [];
 
+	#validIndex (index) {
+		if (!this._el.swatches.children.length) {
+			console.warn("There are no colors to work with.");
+			return false;
+		}
+
+		if (index < 0 || index >= this._el.swatches.children.length) {
+			console.warn(`No color with index "${ index }". The index should be from 0 up to but not including ${ this._el.swatches.children.length }.`);
+			return false;
+		}
+
+		return true;
+	}
+
+	addColor (color, name) {
+		let {name: defaultName, color: defaultColor} = this.defaultColor?.() ?? {};
+
+		name ??= defaultName ?? "New color";
+		color ??= defaultColor ?? this.computedColors.at(-1)?.color ?? new Self.Color("#f06");
```

Comment:
Nice try, but I'd use a middle gray as the ultimate fallback :)

Reply:
> Nice try, but I'd use a middle gray (per OKLab, even if specified as a hex) as the ultimate fallback :)

I was thinking of an easter egg (sort of). 🙃 But yeah, a middle gray would be a good default.

> Also, more performant to specify a color space and array of coords than rely on parsing.

That's a good point. Thanks!